Fixed #5425: Redesign fractions landing page

unit-00 opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include 
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue 
  - when this PR is merged.
  -->
Fixes #5425 
This is a PR for the fractions landing page redesign according to the mock up.

The desktop version looks similar to the mock up and below is a sample of the mobile version.
![issue-5425](https://user-images.githubusercontent.com/10623811/44132635-bd226616-a00f-11e8-9894-4039d9b37de8.gif)



## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes.
- [x] The PR explanation includes the words "Fixes #bugnum: ...".
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python scripts/pre_commit_linter.py` and `bash scripts/run_frontend_tests.sh`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [ ] The PR is assigned to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue.

unit-00 committed:
Change descriptions, change coloring, and also change font size, add explore lessons button

unit-00 committed:
Move h1 titles into flexbox and shifted left to align with h2 titles

unit-00 committed:
Change images to mock and also reduce space between the title and text

unit-00 committed:
Add image files for fraction page

unit-00 committed:
Merge remote-tracking branch 'upstream/develop' into issue-5425

unit-00 committed:
Merge remote-tracking branch 'upstream/develop' into issue-5425

unit-00 committed:
Redesign fractions landing page, add video directory to asset

unit-00 committed:
Remov autoplay from videos and remove placeholder image

unit-00 committed:
Add video type back

unit-00 committed:
Add mp4 to prelint checker

unit-00 committed:
Merge remote-tracking branch 'upstream/develop' into issue-5425

seanlip commented:
Thanks @unit-00! Let me start with UI notes first, based on your very helpful screen animation.

- Please read the original email carefully and make sure that all amendments to the Figma mocks are addressed. In particular, the "pizza"-looking cake should be removed from the first graphic.
- As you scroll down, the card in the second section seems to disappear?
- The bottom "Get Started" button does not seem centralized properly.

Thanks!

seanlip commented on core/templates/dev/head/pages/landing/fractions/Fractions.js:
```diff
@@ -23,6 +23,8 @@ oppia.controller('Fractions', [
       siteAnalyticsService, UrlInterpolationService) {
     $scope.getStaticImageUrl = UrlInterpolationService.getStaticImageUrl;
 
+    $scope.getStaticAssetUrl = UrlInterpolationService.getStaticAssetUrl;
```
For parity with image, let's define getStaticVideoUrl similarly in UrlInterpolationService and add unit tests for it.

unit-00 replied:
Will definitely add this, along with the unit test.

seanlip commented:
(I only had one comment. On the whole, this looks pretty good. Thanks @unit-00!)

unit-00 commented:
Hi @seanlip ,

- I completely misunderstood what you meant on the pizza. I'll definitely remove it.
- Thank you for catching this. I'm unsure of the reason why the card is disappearing, I'll look into it.
- I'll be sure to fix this up.

unit-00 committed:
Merge remote-tracking branch 'upstream/develop' into issue-5425

unit-00 committed:
Remove pizza, center buttons on last section, and add test for getStaticVideoUrl

unit-00 committed:
Fix unclose bracket in UrlInterpolationService.js

unit-00 committed:
Change getStaticAssetUrl to getStaticVideoUrl

unit-00 commented:
Hey @seanlip ,

I've removed the pizza, centered the buttons, added getStaticVideoUrl and its respective tests, but I think the white part disappearing might be a gif issue. 

Everything else seems to be working the way that it is on both desktop and mobile.

seanlip commented on core/templates/dev/head/pages/landing/fractions/landing_page_teacher.html:
```diff
@@ -8,21 +8,21 @@
           <div class="row">
             <div class="col-sm-6 col-sm-push-6 oppia-fractions-landing-image-div" style="height: auto">
-              <img ng-src="<[getStaticImageUrl('/fractions_landing/student/intro_matthew.svg')]>" class="oppia-fractions-landing-image" style="width: 60%" alt="">
+              <img ng-src="<[getStaticImageUrl('/fractions_landing/teachers/matthew_pizza.png')]>" class="oppia-fractions-landing-image" style="width: 60%" alt="">
```
Can we name this image something that doesn't have "pizza" in it?

seanlip commented:
Hi @unit-00, thanks! Just found a couple of additional UI issues when testing:

- The video isn't loading for me. I think the case of "fractions" is wrong in the video filename?
- On narrow screens, sometimes the text falls off the screen -- see "Easy-to-follow lessons" in the first screenshot below.
- There are some CSS issues with the text on the buttons ("Get Started", "Learn More") being much too small on narrow screens. See both screenshots below -- you need to squint to see what's on the buttons:

![fractions-test](https://user-images.githubusercontent.com/10575562/44323909-a7666780-a408-11e8-9fae-c157541afb61.png)

![fractions-test-2](https://user-images.githubusercontent.com/10575562/44323910-a8979480-a408-11e8-82ad-86df720a68b1.png)

Would it be possible to fix these? I think it should be fully ready after that.

Thanks!

unit-00 committed:
rename Fractions_video.mp4 in cache to fractions_video.mp4

unit-00 committed:
rename matthew_pizza.png to matthew_paper.png

unit-00 committed:
"Address review issues: change button and title size for different view
ports."

unit-00 committed:
"Increase button size even more to be easier on the eyes"

unit-00 committed:
Merge remote-tracking branch 'upstream/develop' into issue-5425

unit-00 committed:
"Fix lint issue"

unit-00 committed:
"Take out % on 0 for css styling to better fit lint"

unit-00 commented:
Hey @seanlip ,

Thank you for catching all these issues, I believe I have addressed them! Please look over the changes. 

Is the video working for you now as well? I wanted to be sure because I realize the file uploaded to git was before I renamed it.

seanlip commented on core/templates/dev/head/pages/landing/fractions/landing_page_teacher.html:
```diff
@@ -147,14 +147,14 @@ <h2 class="oppia-fractions-landing-h2" style="color: #242424">Students are guide
             </div>
             <div class="col-sm-6">
-              <h1 class="oppia-fractions-landing-h1" style="color: #FFFFFF; text-align: right; padding-right: 80px; white-space: nowrap;">Easy-to-follow lessons</h1>
+              <h1 id="easyToFollow" class="oppia-fractions-landing-h1" style="color: #FFFFFF; text-align: right; padding-right: 80px;">Easy-to-follow lessons</h1>
```
We don't use ids in general for CSS. Use a class instead, following the existing style.

unit-00 replied:
I'll definitely change it to class to comply with the coding style.

seanlip commented on core/templates/dev/head/pages/landing/fractions/landing_page_teacher.html:
```diff
@@ -257,6 +257,10 @@ <h2 class="oppia-fractions-landing-h2 oppia-fractions-landing-centered-h2" style
     }
+
+    #easyToFollow {
+        white-space: nowrap;
```
All indentation should be by 2 spaces.

unit-00 replied:
ditto

seanlip commented on core/templates/dev/head/pages/landing/fractions/landing_page_teacher.html:
```diff
@@ -382,30 +386,84 @@ <h2 class="oppia-fractions-landing-h2 oppia-fractions-landing-centered-h2" style
+        font-size: 1.6em;
         margin-left: 10%;
+        margin-right: 20%;
+        padding-left: 0 !important;
```
Here and elsewhere: we generally avoid using !important in CSS rules, since it overrides the natural selection order. Could you please find a way to achieve the desired visual effect, but without using !important?

See CSS style guide [here](https://github.com/oppia/oppia/wiki/Coding-style-guide#css).

unit-00 replied:
Oh definitely, I think I can achieve the desired visual effect without the extra media query.

seanlip reviewed (commented):
The buttons look better, thanks!

The aspect ratio of the video seems wrong, though. Could we edit the video to remove the black sides, and choose an aspect ratio for the container to match? The black sides are quite jarring against the green, and it seems like it should be possible for the video to be bigger so that it's more easily readable.

unit-00 committed:
"Remove extra media query, change button size so it is big enough on all
screen sizes, made sure titles are aligned"

unit-00 committed:
Merge remote-tracking branch 'upstream/develop' into issue-5425

unit-00 committed:
Remove video's black bar, shifted the 'easy-to-follow' section downward, fix 'easy-to-follow' overflow

unit-00 committed:
Merge remote-tracking branch 'upstream/develop' into issue-5425

unit-00 committed:
Align get-started and learn-more buttons with the text above

unit-00 commented:
Hi @seanlip ,

I believe I've address the issues along with making some additional change to the overall look of the page. 

Please look them over!

apb7 commented on scripts/pre_commit_linter.py:
```diff
@@ -221,7 +221,7 @@
     'integrations/*', 'integrations_dev/*', '*.svg', '*.gif',
     '*.png', '*.zip', '*.ico', '*.jpg', '*.min.js',
-    'assets/scripts/*', 'core/tests/data/*', '*.mp3')
+    'assets/scripts/*', 'core/tests/data/*', '*.mp3', '*.mp4')
```
LGTM for the changes made in this file only. Thanks!

seanlip commented on core/templates/dev/head/pages/landing/fractions/landing_page_teacher.html:
```diff
@@ -8,21 +8,21 @@
-              <button class="btn oppia-fractions-landing-get-started" ng-click="onClickGetStartedButton('teacher')">Get Started</button>
+            <div class="col-sm-6 col-sm-pull-6" style="z-index:20">
+              <h1 class="oppia-fractions-landing-h1">Fractions just got easier</h1>
+              <h2 class="oppia-fractions-landing-h2" style="padding-right:15px;">Get your students and kids started with our free, effective math lessons.</h2>
```
Nit: add space after ":" in the style attr.

seanlip commented on core/templates/dev/head/pages/landing/fractions/landing_page_teacher.html:
```diff
@@ -8,21 +8,21 @@
+            <div class="col-sm-6 col-sm-pull-6" style="z-index:20">
+              <h1 class="oppia-fractions-landing-h1">Fractions just got easier</h1>
+              <h2 class="oppia-fractions-landing-h2" style="padding-right:15px;">Get your students and kids started with our free, effective math lessons.</h2>
+              <button class="btn oppia-fractions-landing-get-started" ng-click="onClickGetStartedButton('teacher')" style="left:10px;">Get Started</button>
```
Nit: add space after ":" in the style attr.

seanlip commented:
Thanks @unit-00, it's looking good! Just one comment:

![screenshot from 2018-08-29 02-23-11](https://user-images.githubusercontent.com/10575562/44778698-86052a00-ab32-11e8-89a9-ab1e973c6eaf.png)

Seems like the bottom button is not centered properly. (And the top one too, for that matter, comparing with the text just above it?)

Also, the text on the buttons is now a little too big. Could we make it a little smaller? It should probably be just a bit bigger than the "Get your students ... lessons" text. Ditto for the buttons at the bottom of the page.

Finally, could we embed the video in a phone outline, something like this? https://pixabay.com/en/smartphone-icon-modern-symbol-1557796/

Thanks!

unit-00 commented:
Hey @seanlip ,

quick feedback. How does this look from the text size above to the button's text size.

![image](https://user-images.githubusercontent.com/10623811/44812442-0c9a2580-ab8c-11e8-9989-3c5a02fcaffc.png)

Similarly,

![image](https://user-images.githubusercontent.com/10623811/44812366-e96f7600-ab8b-11e8-8750-64808f618779.png)

seanlip commented:
Looks good to me, thanks!

unit-00 committed:
remove left shifting, change button text size to be proportionate to text above, add a new media query, put text description in a container

seanlip reviewed (commented):
Oh wow, I just realized I hadn't submitted this review yet, sorry! Doing it now.

Also, any chance we could get this in soon? Want to include it in the September release...

Thanks!

unit-00 committed:
add smartphone border

unit-00 committed:
Merge remote-tracking branch 'upstream/develop' into issue-5425

unit-00 commented:
Hey @seanlip ,

I great apologize for the delay. I got a bit too frustrated with making sure the video align properly in the smartphone image.

I wasn't sure how to exactly arrange the video inside the image so I made a border to look like a smartphone instead. 

<img width="349" alt="messages image 1225722269" src="https://user-images.githubusercontent.com/10623811/45187703-81054200-b1e6-11e8-82e0-952df76c97ca.png">

unit-00 committed:
fill in gap between video and border

seanlip commented on core/templates/dev/head/pages/landing/fractions/landing_page_teacher.html:
```diff
@@ -335,54 +368,113 @@ <h2 class="oppia-fractions-landing-h2 oppia-fractions-landing-centered-h2" style
     .oppia-fractions-landing-image-mobile {
       display: none;
     }
+    .video-frame {
```
Could you please prefix this class name with .oppia-fractions-landing-... ? Ditto for anything below this one as well (text-box, easy-to-follow, etc.) -- want to make sure the CSS is well-scoped and only applies to this page.

unit-00 replied:
Oh ops, didn't see these reviews. I'll definitely fix these.

seanlip commented on core/templates/dev/head/pages/landing/fractions/landing_page_teacher.html:
```diff
@@ -335,54 +368,113 @@ <h2 class="oppia-fractions-landing-h2 oppia-fractions-landing-centered-h2" style
       display: none;
     }
+    .video-frame {
+      position: relative;
```
For each CSS rule please alphabetize (e.g. background then border then border-bottom-width etc.) Ditto below.

unit-00 replied:
Ditto.

seanlip commented on core/templates/dev/head/pages/landing/fractions/landing_page_teacher.html:
```diff
@@ -335,54 +368,113 @@ <h2 class="oppia-fractions-landing-h2 oppia-fractions-landing-centered-h2" style
       .oppia-fractions-landing-h2 {
         font-size: 1em;
-        text-align: left !important;
+        text-align: center !important;
```
Is it possible at all to get the desired effect without using !important, here and above? In general we try to avoid it.

unit-00 replied:
This was actually from whoever first designed this. Should I take out all the !important? I edited it a bit to change certain stylings.

seanlip reviewed (changes requested):
Hi @unit-00, just took a look, and I think it looks amazing! I love what you did with the phone, it looks great. Sorry to hear that it was a bit frustrating though.

I left a few code comments! UI comments coming shortly.

seanlip commented:
A couple of minor UI things:
- When you hover over the video in such a way that clicking would play it, the mouse pointer should look like something that can click (I think a hand, usually?).
- On larger widths, the text between the buttons at the bottom is not centered -- in the screenshot below, you'll see that the text juts out to the right more than the left:

![screenshot from 2018-09-07 03-29-03](https://user-images.githubusercontent.com/10575562/45214109-68307700-b24e-11e8-89f6-c2b33640a200.png)
- Also, could we change the last exclamation mark (after "Library") to a period? Two exclamation marks in one go seems a bit much IMO.

Otherwise, it looks really nice, thanks a lot! Just so you know, the release cut for September has been delayed to this weekend, so if you could finish this and submit it by tomorrow or early Saturday, then it can go out in the September release.

Thanks!

unit-00 committed:
center library text and align get started and explore buttons in the last section

unit-00 committed:
Merge remote-tracking branch 'upstream/develop' into issue-5425

unit-00 commented:
Hi @seanlip ,

I've centered the the library text and also change the exclamation mark to the period. 

~~However it might be a bit harder to change the cursor for the html5 video.
Based on these SO posts, [SO 0](https://stackoverflow.com/questions/48693832/need-to-show-html5-video-controls-on-hover-or-play) and [SO 1](https://stackoverflow.com/a/48727816), it seems as though the controls are set by the browser.~~

~~I've been trying but it's only allowing me to change the cursor for the timeline, the volume, and the fullscreen button, not the play/pause, I'm going to see if I can do it some other way.~~

Edit: I was able to change everything to pointers upon hover and got around the play button issue. I've also took out the !important css stylings.

Please look over the code and see if I've address everything. 

Thank you.

unit-00 committed:
add oppia-fractions-landing prefix and alphabetize css rules

unit-00 committed:
remove easy-to-follow class to add styling to overall h1, remove important in css styling and adjusting styling in response to removing important

unit-00 committed:
"made font size smaller a bit"

unit-00 committed:
Change cursor to pointer on video hover, and video plays when clicking
on it instead of just play button

unit-00 committed:
Change "video-frame" class to "oppia-fractions-landing-video-frame"

seanlip commented:
Hi @unit-00, this looks awesome, thanks so much! Merging now :) -- it'll be live in mid-Sept once the release has been QA'd and pushed!
